cmd/statusd/main.go

```diff
@@ -209,24 +251,32 @@ Options:
 // haltOnInterruptSignal catches interrupt signal (SIGINT) and
 // stops the node. It times out after 5 seconds
 // if the node can not be stopped.
-func haltOnInterruptSignal(nodeManager common.NodeManager) {
-	signalCh := make(chan os.Signal, 1)
-	signal.Notify(signalCh, os.Interrupt)
-	defer signal.Stop(signalCh)
-	<-signalCh
+func haltOnInterruptSignal(nodeManager common.NodeManager) <-chan struct{} {
+	interruptCh := make(chan struct{})
 
-	log.Println("Got interrupt, shutting down...")
+	go func() {
+		signalCh := make(chan os.Signal, 1)
+		signal.Notify(signalCh, os.Interrupt)
+		defer signal.Stop(signalCh)
+		<-signalCh
 
-	nodeStopped, err := nodeManager.StopNode()
-	if err != nil {
-		log.Printf("Failed to stop node: %v", err.Error())
-		os.Exit(1)
-	}
+		close(interruptCh)
 
-	select {
-	case <-nodeStopped:
-	case <-time.After(time.Second * 5):
-		log.Printf("Stopping node timed out")
-		os.Exit(1)
-	}
+		log.Println("Got interrupt, shutting down...")
+
+		nodeStopped, err := nodeManager.StopNode()
+		if err != nil {
+			log.Printf("Failed to stop node: %v", err.Error())
```

Comment:
Why do we need to call `err.Error()` when using `%v` in the fmt string?

Reply:
No need, my mistake.